bundle/edu.gemini.catalog/src/main/scala/edu/gemini/catalog/image/ImageCatalogClient.scala

```diff
@@ -179,29 +179,14 @@ object ImageCatalogClient {
     def openConnection: Task[ConnectionDescriptor] = Task.delay {
       Log.info(s"Downloading image at $url")
       val connection = url.openConnection()
-      ConnectionDescriptor(Option(connection.getContentType), Option(connection.getContentEncoding))
+      ConnectionDescriptor(Option(connection.getContentType), Option(connection.getContentEncoding), connection.getInputStream)
     }
 
     /**
       * Downloads the url and writes it to a file
       */
-    def writeToTempFile(file: File): Task[Unit] = Task.delay {
-      // Simple old fashioned download
-      val out = new FileOutputStream(file)
-      try {
-        val in = url.openStream()
```

Comment:
This line was opening a new connection even though the caller already has an open one.